### PR TITLE
Update sumo 2019-12

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.1.1.bb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#-------------------------------------------------------------------------------
+
+SRC_URI = "git://github.com/mendersoftware/mender-artifact.git;protocol=https;branch=3.1.x"
+
+# Tag: 3.1.1
+SRCREV = "1bdd3da8203ec274ad4ff9d0bc6de7907372b074"
+
+# Enable this in Betas, not in finals.
+# Downprioritize this recipe in version selections.
+#DEFAULT_PREFERENCE = "-1"
```
Comment (lluiscampos) [inline meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.1.1.bb]: I will fix these in this review also

**Comment by lluiscampos** [comment]
Builds:

- Mender 2.2.1 w/ sumo-next @ [Jenkins 4454](https://mender-jenkins.mender.io/job/mender-builder/4454/)
- Mender 2.1.1 w/ sumo-next @ [Jenkins 4455](https://mender-jenkins.mender.io/job/mender-builder/4455/)

**Comment by lluiscampos** [comment]
> > `test_uboot_automation: Update failed board count for Flash tests.`
> 
> The version of U-Boot used in sumo is closer to the original listed in the source. Although both may work due to statistical wiggle room, I would omit this cherry-pick.

Thanks. Removed this cherry-pick.

**Comment by lluiscampos** [comment]
New builds after fixing unit test builds for mender (see 3e0b3c4)
- 2.2.1 @ [Jenkins 4456](https://mender-jenkins.mender.io/job/mender-builder/4456/)
- 2.1.1 @ [Jenkins 4457](https://mender-jenkins.mender.io/job/mender-builder/4457/)

**Comment by lluiscampos** [comment]
And here, integration test `test_demo_artifact` failed on 2.1.1 for x86 while passing for vexpress and in both on 2.2.1

By the same reasoning than in #878 I think it can be ignored.

**Comment by kacf** [comment]
Indeed, and this test has been known to be unstable in earlier revisions.

**Pull request merged**